code/game/machinery/doors/brigdoors.dm

```diff
@@ -23,147 +23,138 @@
 	var/list/obj/machinery/targets = list()
 
 
-	initialize()
-		..()
+/obj/machinery/door_timer/initialize()
+	..()
 
-		pixel_x = ((src.dir & 3)? (0) : (src.dir == 4 ? WORLD_ICON_SIZE : -WORLD_ICON_SIZE))
-		pixel_y = ((src.dir & 3)? (src.dir ==1 ? WORLD_ICON_SIZE : -WORLD_ICON_SIZE) : (0))
+	pixel_x = ((src.dir & 3)? (0) : (src.dir == 4 ? WORLD_ICON_SIZE : -WORLD_ICON_SIZE))
+	pixel_y = ((src.dir & 3)? (src.dir ==1 ? WORLD_ICON_SIZE : -WORLD_ICON_SIZE) : (0))
 
-		for(var/obj/machinery/door/window/brigdoor/M in all_doors)
-			if (M.id_tag == src.id_tag)
-				targets += M
+	for(var/obj/machinery/door/window/brigdoor/M in all_doors)
+		if (M.id_tag == src.id_tag)
+			targets += M
 
-		for(var/obj/machinery/flasher/F in flashers)
-			if(F.id_tag == src.id_tag)
-				targets += F
+	for(var/obj/machinery/flasher/F in flashers)
+		if(F.id_tag == src.id_tag)
+			targets += F
 
-		for(var/obj/structure/closet/secure_closet/brig/C in brig_lockers)
-			if(C.id_tag == src.id_tag)
-				targets += C
-
-		if(targets.len==0)
-			stat |= BROKEN
-		update_icon()
+	for(var/obj/structure/closet/secure_closet/brig/C in brig_lockers)
+		if(C.id_tag == src.id_tag)
+			targets += C
 
+	if(targets.len==0)
+		stat |= BROKEN
+	update_icon()
 
 //Main door timer loop, if it's timing and time is >0 reduce time by 1.
 // if it's less than 0, open door, reset timer
 // update the door_timer window and the icon
-	process()
-		if(stat & (NOPOWER|BROKEN))
-			return
-		if(src.timing)
-			if(world.time > src.releasetime)
-				src.timer_end() // open doors, reset timer, clear status screen
-				src.timing = 0
-			src.updateUsrDialog()
-			src.update_icon()
-		else
-			timer_end()
+/obj/machinery/door_timer/process()
+	if(stat & (NOPOWER|BROKEN))
 		return
+	if(src.timing)
+		if(world.time > src.releasetime)
+			src.timer_end() // open doors, reset timer, clear status screen
+			src.timing = 0
+		src.updateUsrDialog()
+		src.update_icon()
+	else
+		timer_end()
 
 
 // has the door power sitatuation changed, if so update icon.
-	power_change()
-		..()
-		update_icon()
-		return
+/obj/machinery/door_timer/power_change()
+	..()
```

Comment:
I slashed the update_icon here

Reply:
why

Reply:
Mistake
Probably removed it while trying to remove the return